### PR TITLE
Document how to scope an operator

### PR DESCRIPTION
**Pull request by awgreene**

### CODE REVIEW

File: docs/operator-scoping.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+>Note: In the case that both a selector and a list of namespaces are provided, the selector is ignored.
+
+#### TargetNamespaces and their realationship to InstallModes
+
+When creating `OperatorGroups` it is important to keep in mind that an operator may not support all namespace configurations. For example, an operator that is designed to run at the cluster level shouldn't be expected to work in an `OperatorGroup` that defines a single targetNamespace.
+
+Operator authors are responsible for defining which `InstallModes` their operator supports within its `ClusterServiceVersion (CSV)`. An InstallMode consists of an InstallModeType field and a boolean Supported field. A CSV's spec can contain a set of InstallModes of four distinct InstallModeTypes:
+
+* **OwnNamespace**: If supported, the operator can be a member of an OperatorGroup that selects its own namespace
```
Comment (dmesser) [inline docs/operator-scoping.md]: The explanation around "operator can be a member of an OperatorGroup that selects..."  is too scientific. What it really comes down is that the Operator supports / does not support watching

- only the namespace it is deployed in
- a single given namespace
- a comma-separated list of namespaces
- the entire cluster

This paired with the OperatorGroup decides

- which namespace(s) the Operator is configured to watch
- in which namespace(s) the Operator is discoverable / useable by the cluster users

File: docs/operator-scoping.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+metadata:
+  name: scoped-permissions-operator-group
+  namespace: own-namespace
+spec:
+  serviceAccountName: member-operator-servicee-account
+  targetNamespaces:
+  - own-namespace
+```
+
+Any operator tied to this OperatorGroup will now be confined to the permission(s) granted to the specified ServiceAccount. If the operator asks for permission(s) that are outside the scope of the ServiceAccount the install will fail with appropriate error(s). When OLM installs the operator the following will happen:
```
Comment (kevinrizza) [inline docs/operator-scoping.md]: It seems like there should be another section below this based on how the previous sentence ended. Or am I misreading?

File: docs/operator-scoping.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+metadata:
+  name: all-namespaces-operator-group
+  namespace: own-namespace
+```
+
+### Scoping Member Operator Permissions
+
+When creating an `OperatorGroup`, cluster admins may specify a `ServiceAccount` that defines the set of permissions that may be granted to all member operators. OLM will ensure that when an operator is installed its privileges are confined to that of the `ServiceAccount` specified.
+
+As a result a cluster-admin can limit an operator to a pre-defined set of RBAC rules. The Operator will not be able to do anything that is not explicitly permitted by these permissions. This enables self-sufficient installation of Operators by non-cluster-admin users with a limited scope.
```
Comment (dmesser) [inline docs/operator-scoping.md]: Here we need to point to another doc explaining this a little more.

File: docs/operator-scoping.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (awgreene) [reply docs/operator-scoping.md]: Perhaps `is scoped to` would be a better choice.

File: docs/operator-scoping.md (hunk 2 above)
Comment (awgreene) [reply docs/operator-scoping.md]: My mistake, the final sentence should not exist.

File: docs/operator-scoping.md (hunk 3 above)
Comment (awgreene) [reply docs/operator-scoping.md]: Should we add the following:

>An example of scoping an operator can be found [here]()?

Where `here`  points to `How do I scope down an Operator` which is defined as an advanced use case?

File: docs/operator-scoping.md (hunk 1 above)
Comment (awgreene) [reply docs/operator-scoping.md]: Okay - I will update this portion of the PR to look similar to:
```
Operator authors are responsible for defining which `InstallModes` their operator supports within its `ClusterServiceVersion (CSV)`. There are four `InstallModes` that an operator can support:

* **OwnNamespace**: If supported, the operator can be configured to watch for events in the namespace it is deployed in.
* **SingleNamespace**:If supported, the operator can be configured to watch for events in a single namespace.
* **MultiNamespace**: If supported, the operator can be configured to watch for events in more than one namespace.
* **AllNamespaces**: If supported, the operator can be configured to watch for events in all namespaces.
```

File: docs/operator-scoping.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
+# Operator Scoping
```
[the remark this answers is not in the document]
Comment (awgreene) [reply docs/operator-scoping.md]: I have added the `Configuring the End User Experience` section in response to this comment.

[the remark this answers is not in the document]
Comment (awgreene) [reply docs/operator-scoping.md]: This is the resolved in the latest version of the commit - please take a moment to review.

File: docs/operator-scoping.md (hunk 1 above)
Comment (awgreene) [reply docs/operator-scoping.md]: This has been resolved in the latest version of the commit - please take a moment to review.

File: docs/operator-scoping.md (hunk 3 above)
Comment (awgreene) [reply docs/operator-scoping.md]: This has been resolved in the latest version of the commit - please take a moment to review.

File: docs/operator-scoping.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+A cluster admin can add an operator to an `OperatorGroup` by creating a `Subscription` in the same namespace. An operator can be added and removed from an `OperatorGroup` at anytime.
+
+Keep in mind that this process can be repeated for any number of `OperatorGroups`. This means that a cluster admin can decide for a set of operator to watch for events in one set of namespaces while defining a seperate set of operators that watch for events in a seperate set of namespaces.
+
+### Enabling a User to use an Operator
+
+Once a cluster admin has decided which operators are available on a cluster, it is now time to decide which user(s) may take advantage of an available operator.
+
+Users interact with operators by creating a resource in a namespace that an operator is watching. As such, by tuning RBAC privilidges a cluster admin has complete control over the set of users that may interact with an operator along with the set of APIs that user may take advantage of.
```
Comment (dinhxuanvu) [inline docs/operator-scoping.md]: nit: privileges

File: docs/operator-scoping.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+
+For example, if an operator were avialble that offered two APIs, a cluster admin may provide a user with full RBAC priviledges over one API but not grant the user any priviledges with the second API.
```
Comment (dinhxuanvu) [inline docs/operator-scoping.md]: nit: available
nit: privileges (double)

File: docs/operator-scoping.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+Cluster admins may not know in advance which namespaces member operators should be scoped to. In this case, it may make sense to use a label selector to identify which namespaces permissions should be granted in. A namespace selector can be defined for an `OpearatorGroup` like so:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha2
+kind: OperatorGroup
+metadata:
+  name: my-group
+  namespace: my-namespace
+spec:
+  selector:
```
[the remark this answers is not in the document]
Comment (awgreene) [reply docs/operator-scoping.md]: This is a wonderful point Vu, thanks.

File: docs/operator-scoping.md [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+>Note: In the case that both a selector and a list of namespaces are provided, the selector is ignored.
+
+#### TargetNamespaces and their realationship to InstallModes
+
+When creating `OperatorGroups` it is important to keep in mind that an operator may not support all namespace configurations. For example, an operator that is designed to run at the cluster level shouldn't be expected to work in an `OperatorGroup` that defines a single targetNamespace.
+
+Operator authors are responsible for defining which `InstallModes` their operator supports within its `ClusterServiceVersion (CSV)`. There are four `InstallModes` that an operator can support:
+
+* **OwnNamespace**: If supported, the operator can be configured to watch for events in the namespace it is deployed in.
+* **SingleNamespace**:If supported, the operator can be configured to watch for events in a single namespace.
```
[the remark this answers is not in the document]
Comment (awgreene) [reply docs/operator-scoping.md]: Fair, I will update the description to:
> If supported, the operator can be configured to watch for events in a single namespace that the operator is not deployed in.

**Comment by awgreene** [comment]
@dmesser @kevinrizza @dinhxuanvu I have addressed all available comment. Please re-review.

**Comment by kevinrizza** [comment]
lgtm

**Pull request merged**